nominal/experimental/logging/nominal_log_handler.py

```diff
@@ -0,0 +1,114 @@
+import atexit
+import datetime
+import logging
+import queue
+import threading
+import time
+
+from nominal.core.dataset import Dataset
+from nominal.core.log import LogPoint
+
+DEFAULT_LOG_BATCH_SIZE = 1000
+DEFAULT_LOG_FLUSH_INTERVAL = datetime.timedelta(seconds=5)
+
+
+class NominalLogHandler(logging.Handler):
+    """A custom logging handler that batches log records and sends them to Nominal in a background thread.
+
+    NOTE: to log custom args from a `logger.log(...)` statement, you can pass args as a dictionary via `extras`
+          Example:
+            logger.info("infotainment logs", extra={"nominal_args": {"country": "america", "count": 1234}})
+          This would allow users to see the custom log args within the Nominal log panel.
+
+    NOTE: it is recommended to NOT install this on the root logger, as otherwise, logs that occur during
+          log uploading (e.g. in urllib) will result in an infinite chain of logs being produced to the
+          dataset.
```

Comment:
that's a funny edge case. Is having this logger specifically filter/ignore urllib a workable alternative? (or equivalently, giving a function to register this which adds the filter)